src/BlazorWebView/src/SharedSource/WebView2WebViewManager.cs

```diff
@@ -177,27 +177,46 @@ protected override void NavigateCore(Uri absoluteUri)
 		{
 			_ = Dispatcher.InvokeAsync(async () =>
 			{
-				await _webviewReadyTask;
-				_webview.Source = absoluteUri;
+				var isWebviewInitialized = await _webviewReadyTask;
+
+				if (isWebviewInitialized)
+				{
+					_webview.Source = absoluteUri;
+				}
 			});
 		}
 
 		/// <inheritdoc />
 		protected override void SendMessage(string message)
 			=> _webview.CoreWebView2.PostWebMessageAsString(message);
 
-		private async Task InitializeWebView2()
+		private async Task<bool> TryInitializeWebView2()
 		{
 			var args = new BlazorWebViewInitializingEventArgs();
 #if WEBVIEW2_MAUI
 			_blazorWebViewHandler.VirtualView.BlazorWebViewInitializing(args);
-			_coreWebView2Environment = await CoreWebView2Environment.CreateWithOptionsAsync(
-				browserExecutableFolder: args.BrowserExecutableFolder,
-				userDataFolder: args.UserDataFolder,
-				options: args.EnvironmentOptions)
-				.AsTask()
-				.ConfigureAwait(true);
-			await _webview.EnsureCoreWebView2Async();
+
+			try
+			{
+				_coreWebView2Environment = await CoreWebView2Environment.CreateWithOptionsAsync(
+					browserExecutableFolder: args.BrowserExecutableFolder,
+					userDataFolder: args.UserDataFolder,
+					options: args.EnvironmentOptions)
+					.AsTask()
+					.ConfigureAwait(true);
+			}
+			catch (FileNotFoundException)
+			{
+				// This type of exception is thrown when the WebView2 Runtime is not installed.
+				return false;
+			}
+			finally
+			{
+				// This method needs to be invoked even if the WebView2 Runtime is not installed,
+				// since it is reponsible for creating the warning label and WebView2 Runtime
+				// download link.
+				await _webview.EnsureCoreWebView2Async();
+			}
```

Comment:
I expect this pattern is fine, but it does mean now that if `CreateWithOptionsAsync` throws something other than a `FileNotFoundException`, we'll now still call `EnsureCoreWebView2Async` whereas previously we wouldn't have done.

No idea if that could confuse matters (e.g., by surfacing a different and less obvious exception).

One option would be to eliminate the `finally` block and have two calls to `EnsureCoreWebView2Async` - one inside the `catch`, and the other after the `try`/`catch` entirely.